### PR TITLE
added a function that makes sure all the needed properties are there

### PR DESCRIPTION
**Pull request by freshie**
added a function that makes sure all the needed properties are in a jobDoc and update the code to call that when pull a job doc from the database

### CODE REVIEW

File: state-conductor-modules/src/main/ml-modules/root/state-conductor/state-conductor.sjs [hunk 1]
```diff
@@ -858,6 +844,28 @@ function isLatestTemporalDocument(uri) {
...
+    flowName: null,
+    flowStatus: null,
+    flowState: null,
+    uri: null,
+    database: null,
+    modules: null,
+    createdDate: null,
+    context: {},
+    provenance: []
+  };
```
Comment (aclavio) [inline state-conductor-modules/src/main/ml-modules/root/state-conductor/state-conductor.sjs]: Should we also initialize an empty "errors" array?  We could eliminate the check prior to adding error blocks if we did the initialization here.

**Comment by freshie** [comment]
> createStateConductorJob

I've add the scaffoldJobDoc to the createStateConductorJob but I dont want to use the values from createStateConductorJob  because I dont want values createdDate to be added. I'd rather have it error out with the null value then not error out but not do what they want.

File: state-conductor-modules/src/main/ml-modules/root/state-conductor/state-conductor.sjs (hunk 1 above)
Comment (freshie) [reply state-conductor-modules/src/main/ml-modules/root/state-conductor/state-conductor.sjs]: Thats a good idea. I'll add it.

Comment (freshie) [reply state-conductor-modules/src/main/ml-modules/root/state-conductor/state-conductor.sjs]: Added this in

**Pull request merged**